--- origsrc/telepathy-gabble-0.8.7/lib/gibber/gibber-unix-transport.c	2009-10-14 07:29:59.000000000 -0500
+++ src/telepathy-gabble-0.8.7/lib/gibber/gibber-unix-transport.c	2009-10-29 18:56:17.044615900 -0500
@@ -98,8 +98,10 @@ gibber_unix_transport_class_init (
   object_class->dispose = gibber_unix_transport_dispose;
   object_class->finalize = gibber_unix_transport_finalize;
 
+#if !defined(__CYGWIN__)
   /* override GibberFdTransport's read */
   fd_class->read = gibber_unix_transport_read;
+#endif
 }
 
 void
@@ -199,6 +201,9 @@ gibber_unix_transport_send_credentials (
     const guint8 *data,
     gsize size)
 {
+#if defined(__CYGWIN__)
+  return FALSE;
+#else
   int fd, ret;
   struct ucred *cred;
   struct msghdr msg;
@@ -239,10 +244,12 @@ gibber_unix_transport_send_credentials (
     }
 
   return TRUE;
+#endif
 }
 
 #define BUFSIZE 1024
 
+#if !defined(__CYGWIN__)
 static GibberFdIOResult
 gibber_unix_transport_read (GibberFdTransport *transport,
     GIOChannel *channel,
@@ -338,6 +345,7 @@ gibber_unix_transport_read (GibberFdTran
   priv->recv_creds_data = NULL;
   return GIBBER_FD_IO_RESULT_SUCCESS;
 }
+#endif /* ! __CYGWIN__ */
 
 gboolean
 gibber_unix_transport_recv_credentials (GibberUnixTransport *self,
